enterprise/com/src/main/java/org/neo4j/com/PortRangeSocketBinder.java

```
@@ -0,0 +1,74 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.com;
+
+import org.jboss.netty.bootstrap.ServerBootstrap;
+import org.jboss.netty.channel.Channel;
+import org.jboss.netty.channel.ChannelException;
+
+import java.net.InetSocketAddress;
+
+import org.neo4j.helpers.HostnamePort;
+
+public class PortRangeSocketBinder
+{
+    private ServerBootstrap bootstrap;
+    private static final String ALL_INTERFACES_ADDRESS = "0.0.0.0";
+
+    public PortRangeSocketBinder( ServerBootstrap bootstrap )
+    {
+        this.bootstrap = bootstrap;
+    }
+
+    public Connection bindToFirstAvailablePortInRange( HostnamePort serverAddress ) throws ChannelException
+    {
+        int[] ports = serverAddress.getPorts();
+        String host = serverAddress.getHost();
+
+        Channel channel;
+        InetSocketAddress socketAddress;
+        ChannelException lastException = null;
+
+        PortIterator portIterator = new PortIterator( ports );
+        while ( portIterator.hasNext() )
+        {
+            Integer port = portIterator.next();
+            if ( host == null || host.equals( ALL_INTERFACES_ADDRESS ) )
+            {
+                socketAddress = new InetSocketAddress( port );
+            }
+            else
+            {
+                socketAddress = new InetSocketAddress( host, port );
+            }
+            try
+            {
+                channel = bootstrap.bind( socketAddress );
+                return new Connection( socketAddress, channel );
+            }
+            catch ( ChannelException e )
+            {
+                lastException = e;
```

Sorry my bad, I didn't explain it properly. I was trying to suggest something like:
```
if (lastException != null)
{
   lastException = e
}
else
{
  e.addSuppressed(lastException);
  lastException = e;
} 
```

In order not to swallow all the exceptions.